libr/bin/bfile.c

```diff
@@ -407,7 +407,23 @@ static void get_strings_range(RBinFile *bf, RList *list, int min, int raw, ut64
 			return;
 		}
 	}
-	string_scan_range (list, bf, min, from, to, -1, raw, section);
+	int type;
+	const char *enc = bf->rbin->strenc;
```

Comment:
What about extracting it into the separate function?

Reply:
Could be done but right now the code is unique afaik. There's another code structure like it in disasm.c but the enum involved is different. Probably will merge the two enums when the need arises.

Reply:
Also have to decide whether it's best for the encoding to be stored as a string or an enum.